scapy/layers/rtp.py

```diff
@@ -25,16 +25,25 @@
     31: 'H261',          32: 'MPV',
     33: 'MP2T',          34: 'H263' }
 
+
 class RTP(Packet):
-    name="RTP"
-    fields_desc = [ BitField('version', 2, 2),
-                    BitField('padding', 0, 1),
-                    BitField('extension', 0, 1),
-                    BitFieldLenField('numsync', None, 4, count_of='sync'),
-                    BitField('marker', 0, 1),
-                    BitEnumField('payload_type', 0, 7, _rtp_payload_types),
-                    ShortField('sequence', 0),
-                    IntField('timestamp', 0),
-                    IntField('sourcesync', 0),
-                    FieldListField('sync', [], IntField("id",0), count_from=lambda pkt:pkt.numsync) ]
-    
+    name = "RTP"
+    extension_condition = lambda pkt: (pkt.extension == 1)
+    fields_desc = [
+                   BitField('version', 2, 2),
+                   BitField('padding', 0, 1),
+                   BitField('extension', 0, 1),
+                   BitFieldLenField('numsync', None, 4, count_of='sync'),
+                   BitField('marker', 0, 1),
+                   BitEnumField('payload_type', 0, 7, _rtp_payload_types),
+                   ShortField('sequence', 0),
+                   IntField('timestamp', 0),
+                   IntField('sourcesync', 0),
+                   FieldListField('sync', [], IntField("id", 0), count_from=lambda pkt: pkt.numsync),
```

Comment:
I think that it will be nicer to implement the extension header as an independent layer (i.e. RTPExtension) and use guess_payload_class() to decode it when the extension bit is set to 1.